src/common/blockchain.sh

```diff
@@ -1,3 +1,201 @@
+            return 2
+        else
+            echo "Unrecognized error returned:"
+            echo "$OUTPUT"
+            return 1
+        fi  
+    fi
+
+    echo "Successfully installed fabric contract."
+    return 0
+}
+
+function instantiate_fabric_chaincode {
+    CC_ID=$1
+    CC_VERSION=$2
+    CHANNEL=$3
+    INIT_ARGS=$4
+
+    cat << EOF > request.json
+{
+    "chaincode_id": "${CC_ID}",
+    "chaincode_version": "${CC_VERSION}",
+    "chaincode_arguments": [${INIT_ARGS}]
+}
+EOF
+
+    request_url="${BLOCKCHAIN_URL}/api/v1/networks/${BLOCKCHAIN_NETWORK_ID}/channels/${CHANNEL}/chaincode/instantiate"
+
+    echo "Instantiating fabric contract with id '$CC_ID' and version '$CC_VERSION' on channel '$CHANNEL' with arguments '$INIT_ARGS'..."
+
+    OUTPUT=$(
+    do_curl \
+        -X POST \
+        -H 'Content-Type: application/json' \
+        -u "${BLOCKCHAIN_KEY}:${BLOCKCHAIN_SECRET}" \
+        --data-binary @request.json \
+        "${request_url}"
+    )
+    do_curl_status=$?
+
+    rm -f request.json
+
+    if [[ "${OUTPUT}" == *"Failed to establish a backside connection"* || "${OUTPUT}" == *"premature execution"* ]]
+    then
+        echo "Connection problem encountered, delaying 30s and trying again..."
+        sleep 30
+        instantiate_fabric_chaincode "$@"
+        return $?
+    fi
+
+    if [ $do_curl_status -eq 1 ]
+    then
+        echo "Failed to instantiate fabric contract:"
+        if [[ "${OUTPUT}" == *"version already exists for chaincode"* ]]
+        then
+            echo "Chaincode instance already exists with id '${CC_ID}' and version '${CC_VERSION}'"
+            return 2
+        else
+            echo "Unrecognized error returned:"
+            echo "${OUTPUT}"
+            return 1
+        fi
+    fi
+
+    echo "Successfully instantiated fabric contract."
+    return 0
+}
+
+
+function parse_fabric_config {
+    NET_CONFIG_FILE=$1
+
+    echo "Parsing deployment configuration:"
+    cat "$NET_CONFIG_FILE"
+
+    for org in $(jq -r "to_entries[] | .key" "$NET_CONFIG_FILE")
+    do
+        echo "Targeting org '$org'..."
+        authenticate_org "$org"
+        
+        cc_index=0
+        jq -r ".${org}.chaincode[].path" "$NET_CONFIG_FILE" | while read -r CC_PATH
+        do
+            CC_NAME=$(jq -r ".${org}.chaincode[$cc_index].name" "$NET_CONFIG_FILE")
+            CC_FILE="${CC_PATH}/${CC_NAME}.go"
+            CC_INSTALL=$(jq -r ".${org}.chaincode[$cc_index].install" "$NET_CONFIG_FILE")
+            CC_INSTANTIATE=$(jq -r ".${org}.chaincode[$cc_index].instantiate" "$NET_CONFIG_FILE")
+            CC_CHANNELS=$(jq -r ".${org}.chaincode[$cc_index].channels[]" "$NET_CONFIG_FILE")
+            CC_INIT_ARGS=$(jq ".${org}.chaincode[$cc_index].init_args[]" "$NET_CONFIG_FILE")
+
+            # TODO: Integrate with configuration
+            CC_ID="${CC_NAME}"
+            CC_VERSION=$(date '+%Y%m%d%H%M%S')
```

Comment:
We need to be able to associate the version of the chaincode with the build number.   I would like the version to include the build number per our previous discussion. I believe the variable is $BUILD_NUMBER